src/jvm/main/org/jetbrains/kotlinx/lincheck/Injections.kt

```diff
@@ -51,12 +51,17 @@ internal object Injections {
         return t.inTestingCode && !t.inIgnoredSection
     }
 
+    @JvmStatic
+    fun beforeLock(codeLocation: Int) {
```

Comment:
Please explain what this function does and why you need to have both `beforeLock` and `lock`. The same applies to the `wait`s.